### PR TITLE
Use std containers instead uthash

### PR DESCRIPTION
**Pull request by halx99**
Which branch your pull-request should merge into?

- `dev`: Current 2.x BugFixs or Features

- [x] sch update list, use insert_sorted aka SortedList for priority sched

## Checklist before requesting a review
-  [x] I have performed a self-review of my code.
-  [ ] If it is a core feature, I have added thorough tests.
-  [ ] I have checked readme and add important infos to this PR (if it needed).
-  [x] I have added/adapted some tests too.

### CODE REVIEW

File: core/2d/ActionManager.h [hunk 1]
```diff
@@ -36,9 +36,14 @@ THE SOFTWARE.
 
 NS_AX_BEGIN
 
-class Action;
-
-struct _hashElement;
+struct _ActionHandle
```
Comment (j-jorge) [inline core/2d/ActionManager.h]: Identifiers beginning with an underscore followed by an uppercase letter are always reserved in C++. You should drop the underscore here. If the idea is to have some "internal" type I would suggest to use a `detail` namespace.

File: core/base/Scheduler.h [hunk 2]
```diff
@@ -156,6 +156,34 @@ struct _listEntry;
 struct _hashSelectorEntry;
 struct _hashUpdateEntry;
 
+// data structures
+
+// A list double-linked list used for "updates with priority"
```
Comment (j-jorge) [inline core/base/Scheduler.h]: There are two many "lists" in this sentence: A ~list~ double-linked list used for "updates with priority"

File: core/base/Vector.h [hunk 3]
```diff
@@ -255,6 +255,12 @@ class Vector
         return _data[index];
     }
 
+    T operator[](size_t index) const
+    {
+        AXASSERT(index >= 0 && index < size(), "index out of range in getObjectAtIndex()");
```
Comment (j-jorge) [inline core/base/Vector.h]: `index >= 0` is always true with size_t.

File: thirdparty/yasio/pod_vector.hpp [hunk 4]
```diff
@@ -162,6 +162,43 @@ class pod_vector {
     }
     return _Where;
   }
+  iterator insert(iterator _Where, const value_type& _Val)
```
Comment (j-jorge) [inline thirdparty/yasio/pod_vector.hpp]: Again here, all these _Identifiers cause undefined behavior.

File: core/base/Scheduler.h (hunk 2 above)
Comment (halx99) [reply core/base/Scheduler.h]: the comments should be remove

File: core/base/Scheduler.h [hunk 5]
```diff
@@ -32,10 +32,10 @@ THE SOFTWARE.
 #include <functional>
 #include <mutex>
 #include <set>
-
+#include <list>
```
Comment (halx99) [inline core/base/Scheduler.h]:
```suggestion
```

**Pull request merged**